server/core/namespaces.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+// Namespace defines two things:
+// 1. relation between a name and several tables
+// 2. relation between a name and several stores
+// It is used to bind tables with stores
+type Namespace struct {
```

Comment:
How about moving it to `server/table_namespace_classifier.go`? Since it is only used by `server` and `server/api` package.